modules/photo/src/calibrate.cpp

```diff
@@ -57,77 +56,104 @@ class CalibrateDebevecImpl : public CalibrateDebevec
         samples(_samples),
         lambda(_lambda),
         random(_random),
-        w(tringleWeights())
+        w(triangleWeights())
     {
     }
 
     void process(InputArrayOfArrays src, OutputArray dst, InputArray _times)
     {
         CV_INSTRUMENT_REGION()
 
+        // check inputs
         std::vector<Mat> images;
         src.getMatVector(images);
         Mat times = _times.getMat();
 
         CV_Assert(images.size() == times.total());
         checkImageDimensions(images);
         CV_Assert(images[0].depth() == CV_8U);
+        CV_Assert(times.depth() == CV_32F);
```

Comment:
Oh there's no need. Although I expect images to be 1 or 3 channels, the code should work for any number of channels (each processed separately).
The output will be 256x1 matrix with the same number of channels as the input images.